syntax = "proto3";

package eco-base.governance;

service YAMLGovernanceService {
  rpc GenerateManifest  (GenerateManifestRequest)  returns (GenerateManifestResponse);
  rpc ValidateManifest  (ValidateManifestRequest)  returns (ValidateManifestResponse);
  rpc RegisterService   (RegisterServiceRequest)   returns (RegisterServiceResponse);
  rpc GetVectorAlignment(VectorAlignmentRequest)   returns (VectorAlignmentResponse);
}

message GenerateManifestRequest {
  string module_name    = 1;
  string image          = 2;
  int32 replicas        = 3;
  repeated int32 ports  = 4;
  repeated string depends_on = 5;
  string target_system  = 6;
  string schema_path    = 7;
}

message GenerateManifestResponse {
  string qyaml_content  = 1;
  string unique_id       = 2;
  bool valid             = 3;
  repeated string warnings = 4;
}

message ValidateManifestRequest {
  string qyaml_content  = 1;
  bool strict            = 2;
}

message ValidateManifestResponse {
  bool valid                    = 1;
  repeated ValidationError errors = 2;
  GovernanceMetadata metadata   = 3;
}

message ValidationError {
  string path    = 1;
  string message = 2;
  string severity = 3;
}

message GovernanceMetadata {
  string unique_id        = 1;
  string target_system    = 2;
  string schema_version   = 3;
  string generated_by     = 4;
  string owner            = 5;
  repeated string compliance_tags = 6;
}

message RegisterServiceRequest {
  string service_name     = 1;
  string endpoint         = 2;
  string discovery_protocol = 3;
  string health_check_path = 4;
  int32 registry_ttl      = 5;
}

message RegisterServiceResponse {
  string service_id       = 1;
  bool registered         = 2;
  string registry_endpoint = 3;
}

message VectorAlignmentRequest {
  string service_id       = 1;
  string alignment_model  = 2;
  int32 target_dim        = 3;
}

message VectorAlignmentResponse {
  repeated float coherence_vector = 1;
  repeated string function_keywords = 2;
  string contextual_binding = 3;
  float alignment_score   = 4;
}